simtools/corsika/corsika_output.py

```diff
@@ -0,0 +1,1002 @@
+        """
+        Get the distribution of the number of photons, including the telescopes indicated by
+        `self.telescope_indices` or all telescopes if `self.telescope_indices` is None.
+
+        Parameters
+        ----------
+        bins: float
+            Number of bins for the histogram.
+        range: 2-tuple
+            Tuple to define the range of the histogram.
+
+        Returns
+        -------
+        numpy.array
+            Number of photons per event.
+        numpy.ndarray
+            The values (counts) of the histogram.
+        """
+
+        num_of_photons_per_event_per_telescope = self.get_num_photons_per_event_per_telescope()
+        if self.telescope_indices is None:
+            num_photons_per_event = np.sum(num_of_photons_per_event_per_telescope, axis=1)
+        else:
+            num_photons_per_event = np.sum(
+                num_of_photons_per_event_per_telescope[self.telescope_indices], axis=1
+            )
+        hist, edges = np.histogram(num_photons_per_event, bins=bins, range=range)
+        return edges, hist
+
+    @property
+    def num_photons_per_event(self):
+        """
+        The number of photons per event.
+
+        Returns
+        -------
+        numpy.array
+            Number of photons per event.
+        """
+        num_of_photons_per_event_per_telescope = self.get_num_photons_per_event_per_telescope()
+        self._num_photons_per_event = np.sum(
+            np.array(num_of_photons_per_event_per_telescope), axis=0
+        )
+        return self._num_photons_per_event
+
+    @property
+    def total_num_photons(self):
+        """
+        The total number of photons.
+
+        Returns
+        -------
+        float
+            Total number photons.
+        """
+        self._total_num_photons = np.sum(self.num_photons_per_event)
+        return self._total_num_photons
+
+    @property
+    def telescope_positions(self):
+        """
+        The telescope positions.
+
+        Returns
+        -------
+        numpy.ndarray
+            x, y and z positions of the telescopes and their radius according to the CORSIKA
+            spherical representation of the telescopes.
+        """
+        return self._tel_positions
+
+    def _get_event_information(self):
+        """
+        Get information from the event header and save into dictionary.
+        """
+        if self._events_information is None:
+            with IACTFile(self.input_file) as f:
+                self._file_header = f.header
+                self._tel_positions = np.array(f.telescope_positions)
+                self.num_telescopes = np.size(self._tel_positions, axis=0)
+                self._events_information = {
+                    key: {"value": [], "unit": None} for key in corsika7_event_header
+                }
+                self.num_events = 0
+                for _, event in enumerate(f):
+                    for key in corsika7_event_header:
```

Comment:
Ok, lets discuss it in person. as you pointed out, the header is also defined here https://github.com/cta-observatory/pycorsikaio/blob/main/corsikaio/subblocks/event_header.py

I am not sure whether corsikaio will continue being supported, but for now it is not a dependency for us. If we want to use their header, corsikaio would have to become a dependency due to a simple "list" of parameters.

Reply:
I don't mind making corsikaio a dependency. It should be supported also in the future, possibly by us.
In this case my question was why not read everything pyeventio reads though instead of defining specific entries.

Reply:
I provided now a possible solution here. Ive been through the corsikaio code and the way I implemented now is to use their definitions but do not complicate much with the functions. In corsikaio they have the definition of the run and events headers but they do not read IACT file. Please have a look and lets discuss in person. Thanks!